### PR TITLE
Discovery

### PR DESCRIPTION
**Pull request by nickbalestra**
- [x] I ran `npm test` for the package I'm modifying
- [x] I used `npm run commit` instead of `git commit`
- [x] Implement further the flavor discovery logic by relying on npm as a registry.
- [x] Use the keyword `create-cycle-app-flavor` to discover community flavors.
- [x] Keep coreFlavors.json for the core flavors.

@geovanisouza92 give it a spin and let me know.

### CODE REVIEW

File: create-cycle-app/index.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ var argv = require('minimist')(process.argv.slice(2))
 var pathExists = require('path-exists')
 var request = require('request')
 var inquirer = require('inquirer')
+var githubUrlFromNpm = require('github-url-from-npm')
```
Comment (geovanisouza92) [inline create-cycle-app/index.js]: Does it really need this dependency? Looking at [the code](https://github.com/hokaccha/github-url-from-npm/blob/master/index.js), it does only make a request to npm API (that we certainly could call), and delegates to [another one](https://github.com/tj/node-github-url-from-git/blob/master/index.js) to just format a url from github.

Besides this, it enforces flavor's authors to publish package on github AND npm in order to be discovered, which seems duplicated effort. (There is a problem in using just npm data?)

File: create-cycle-app/index.js [hunk 2]
```diff
@@ -143,23 +152,30 @@ function discoverFlavors (cb) {
...
-        cb(err)
-      }
-      if (res.statusCode !== 200) {
-        console.error(chalk.red('Flavors request failed with status: ' + res.statusCode))
-        return
-      }
+    var npmContent = JSON.parse(body).rows
+    var communityFlavors = npmContent.map(function (itemFound) {
+      return new Promise(function (resolve, reject) {
+        githubUrlFromNpm(itemFound.key[1], function (err, url) {
```
Comment (geovanisouza92) [inline create-cycle-app/index.js]: Following the comment above, we could just use `request` here as well

**Comment by geovanisouza92** [comment]
It seems a nice work Nick, but with some caveats in the code... I would prefer to avoid dependency bloat to the minimum, if possible.

**Comment by nickbalestra** [comment]
Good points @geovanisouza92, agree 💯 !
Thanks for the quick review

**Comment by nickbalestra** [comment]
@geovanisouza92 Done

**Comment by geovanisouza92** [comment]
Seems ok. Thanks.

**Pull request merged**